tests/unit/plugins/modules/test_file_interface.py

```diff
@@ -0,0 +1,162 @@
+# Copyright: (c) 2024, Dell Technologies
+# Apache License version 2.0 (see MODULE-LICENSE or http://www.apache.org/licenses/LICENSE-2.0.txt)
+"""Unit Tests for file interface module for PowerStore"""
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+import pytest
+# pylint: disable=unused-import
+
+from ansible_collections.dellemc.powerstore.tests.unit.plugins.module_utils.libraries import initial_mock
+from mock.mock import MagicMock
+from ansible_collections.dellemc.powerstore.tests.unit.plugins.module_utils.mock_file_interface_api import MockFileInterfaceApi
+from ansible_collections.dellemc.powerstore.tests.unit.plugins.module_utils.mock_api_exception import MockApiException
+
+from ansible_collections.dellemc.powerstore.plugins.modules.file_interface import PowerStoreFileInterface, FileInterfaceHandler
+from ansible_collections.dellemc.powerstore.tests.unit.plugins.module_utils.libraries.fail_json import FailJsonException, fail_json
+
+
+class TestPowerStoreFileInterface():
+
+    get_module_args = MockFileInterfaceApi.FILE_INTERFACE_COMMON_ARGS
+
+    @pytest.fixture
+    def file_interface_module_mock(self, mocker):
+        mocker.patch(MockFileInterfaceApi.MODULE_UTILS_PATH + '.PowerStoreException', new=MockApiException)
+        file_interface_module_mock = PowerStoreFileInterface()
+        file_interface_module_mock.module = MagicMock()
+        file_interface_module_mock.module.check_mode = False
+        file_interface_module_mock.module.fail_json = fail_json
+        return file_interface_module_mock
+
+    def capture_fail_json_call(self, error_msg, file_interface_module_mock):
```

Comment:
this method will not be required in the test file after we inherit the base class